`revertedWith` not matching errors

as of `2.1.1`, and still in `2.2.0`, tests of the form `.to.be.revertedWith('...')` aren't working properly (they don't catch the actual string from `require`s.

`AssertionError: Expected transaction to be reverted with ..., but other exception was thrown: execution error: revert`

--- a/waffle-chai/package.json
+++ b/waffle-chai/package.json
@@ -35,15 +35,15 @@
     "node": ">=10.0"
   },
   "dependencies": {
-    "ethers": "^4.0.0"
+    "ethers": "^4.0.45"
   },
   "devDependencies": {
+    "@ethereum-waffle/provider": "^2.3.0",
     "@types/chai": "^4.2.3",
     "@types/chai-as-promised": "^7.1.2",
     "@types/mocha": "^5.2.7",
     "@typescript-eslint/eslint-plugin": "^2.6.1",
     "@typescript-eslint/parser": "^2.6.1",
-    "@ethereum-waffle/provider": "^2.3.0",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
     "eslint": "^6.5.1",

--- a/waffle-chai/test/reverted.ts
+++ b/waffle-chai/test/reverted.ts
@@ -25,7 +25,7 @@ describe('INTEGRATION: Matchers: reverted', () => {
   });
 
   it('ThrowAndModify: success', async () => {
-    await expect(matchers.doThrowAndModify()).to.be.rejectedWith('execution error: invalid opcode');
+    await expect(matchers.doThrowAndModify()).to.be.revertedWith('');
   });
 
   it('Revert: success', async () => {
@@ -73,11 +73,11 @@ describe('INTEGRATION: Matchers: revertedWith', () => {
   });
 
   it('Revert with modification: success', async () => {
-    await expect(matchers.doRevertAndModify()).to.be.revertedWith('execution error: revert');
+    await expect(matchers.doRevertAndModify()).to.be.revertedWith('Revert cause');
   });
 
   it('ThrowAndModify: success', async () => {
-    await expect(matchers.doThrowAndModify()).to.be.rejectedWith('execution error: invalid opcode');
+    await expect(matchers.doThrowAndModify()).to.be.revertedWith('');
   });
 
   it('Revert: success', async () => {
@@ -85,7 +85,9 @@ describe('INTEGRATION: Matchers: revertedWith', () => {
   });
 
   it('Revert: fail when different message was thrown', async () => {
-    await expect(expect(matchers.doRevert()).to.be.revertedWith('Different message')).to.be.eventually.rejected;
+    await expect(
+      expect(matchers.doRevert()).to.be.revertedWith('Different message')
+    ).to.be.eventually.rejected;
   });
 
   it('Revert: fail no exception', async () => {
@@ -99,11 +101,15 @@ describe('INTEGRATION: Matchers: revertedWith', () => {
   });
 
   it('Require: fail when no exception was thrown', async () => {
-    await expect(expect(matchers.doRequireSuccess()).to.be.revertedWith('Never to be seen')).to.be.eventually.rejected;
+    await expect(
+      expect(matchers.doRequireSuccess()).to.be.revertedWith('Never to be seen')
+    ).to.be.eventually.rejected;
   });
 
   it('Require: fail when different message', async () => {
-    await expect(expect(matchers.doRequireFail()).to.be.revertedWith('Different message')).to.be.eventually.rejected;
+    await expect(
+      expect(matchers.doRequireFail()).to.be.revertedWith('Different message')
+    ).to.be.eventually.rejected;
   });
 
   it('Not to revert: fail', async () => {

--- a/waffle-cli/buildTestContracts.ts
+++ b/waffle-cli/buildTestContracts.ts
@@ -2,10 +2,11 @@ import {compileAndSave} from '@ethereum-waffle/compiler';
 
 const buildExampleContracts = async () => {
   console.log('Building example contracts...');
-  const sourcesPath = './test/projects/example';
-  const targetPath = './test/example/build';
-  const npmPath = 'node_modules';
-  await compileAndSave({sourcesPath, targetPath, npmPath});
+  const sourceDirectory = './test/projects/example';
+  const outputDirectory = './test/example/build';
+  const nodeModulesDirectory = 'node_modules';
+  const compilerVersion = 'v0.5.9+commit.e560f70d'
+  await compileAndSave({sourceDirectory, outputDirectory, nodeModulesDirectory, compilerVersion});
 };
 
 buildExampleContracts();

--- a/waffle-cli/package.json
+++ b/waffle-cli/package.json
@@ -42,10 +42,10 @@
     "node": ">=10.0"
   },
   "dependencies": {
-    "@ethereum-waffle/provider": "^2.3.2",
     "@ethereum-waffle/chai": "^2.3.0",
     "@ethereum-waffle/compiler": "^2.3.0",
-    "ethers": "^4.0.0"
+    "@ethereum-waffle/provider": "^2.3.2",
+    "ethers": "^4.0.45"
   },
   "devDependencies": {
     "@types/chai": "^4.2.3",

--- a/waffle-compiler/package.json
+++ b/waffle-compiler/package.json
@@ -35,13 +35,15 @@
     "node": ">=10.0"
   },
   "dependencies": {
-    "@types/mkdirp": "^0.5.2",
     "@resolver-engine/imports": "^0.3.3",
     "@resolver-engine/imports-fs": "^0.3.3",
+    "@types/mkdirp": "^0.5.2",
     "mkdirp": "^0.5.1",
-    "solc": "^0.5.10"
+    "solc": "^0.6.3"
   },
   "devDependencies": {
+    "@ethereum-waffle/chai": "^2.3.0",
+    "@ethereum-waffle/provider": "^2.3.0",
     "@types/chai": "^4.2.3",
     "@types/chai-as-promised": "^7.1.2",
     "@types/chai-string": "^1.4.2",
@@ -51,8 +53,6 @@
     "@types/sinon-chai": "^3.2.3",
     "@typescript-eslint/eslint-plugin": "^2.6.1",
     "@typescript-eslint/parser": "^2.6.1",
-    "@ethereum-waffle/chai": "^2.3.0",
-    "@ethereum-waffle/provider": "^2.3.0",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
     "chai-string": "^1.5.0",

--- a/waffle-compiler/src/compileSolcjs.ts
+++ b/waffle-compiler/src/compileSolcjs.ts
@@ -21,7 +21,7 @@ export function compileSolcjs(config: Config) {
   return async function compile(sources: ImportFile[], findImports: (file: string) => any) {
     const solc = await loadCompiler(config);
     const input = buildInputObject(sources, config.compilerOptions);
-    const output = solc.compile(JSON.stringify(input), { imports: findImports });
+    const output = solc.compile(JSON.stringify(input), {imports: findImports});
     return JSON.parse(output);
   };
 }

--- a/waffle-compiler/src/compileSolcjs.ts
+++ b/waffle-compiler/src/compileSolcjs.ts
@@ -21,7 +21,7 @@ export function compileSolcjs(config: Config) {
   return async function compile(sources: ImportFile[], findImports: (file: string) => any) {
     const solc = await loadCompiler(config);
     const input = buildInputObject(sources, config.compilerOptions);
-    const output = solc.compile(JSON.stringify(input), findImports);
+    const output = solc.compile(JSON.stringify(input), { imports: findImports });
     return JSON.parse(output);
   };
 }

--- a/waffle-compiler/test/compiler/compiler.ts
+++ b/waffle-compiler/test/compiler/compiler.ts
@@ -5,7 +5,12 @@ import {inputToConfig} from '../../src/config';
 
 const sourceDirectory = './test/projects/example';
 const outputDirectory = './test/compiler/build';
-const config = inputToConfig({sourceDirectory, outputDirectory});
+const compilerVersion = 'v0.5.9+commit.e560f70d';
+const config = inputToConfig({
+  sourceDirectory,
+  outputDirectory,
+  compilerVersion,
+});
 
 describe('INTEGRATION: Compiler', () => {
   it('compile just compiles', async () => {
@@ -21,7 +26,7 @@ describe('INTEGRATION: Compiler', () => {
   describe('compileAndSave: invalid input', () => {
     const sourceDirectory = './test/projects/invalidContracts'; // tslint:disable-line
     const outputDirectory = './test/projects/build'; // tslint:disable-line
-    const config = inputToConfig({sourceDirectory, outputDirectory});
+    const config = inputToConfig({sourceDirectory, outputDirectory, compilerVersion});
     const expectedOutput = `${'test/projects/invalidContracts/invalid.sol'}:6:14: ` +
       'DeclarationError: Identifier not found or not unique.\n' +
       '  function f(wrongType arg) public {\n             ^-------^\n';

--- a/waffle-compiler/test/compiler/compiler.ts
+++ b/waffle-compiler/test/compiler/compiler.ts
@@ -9,7 +9,7 @@ const compilerVersion = 'v0.5.9+commit.e560f70d';
 const config = inputToConfig({
   sourceDirectory,
   outputDirectory,
-  compilerVersion,
+  compilerVersion
 });
 
 describe('INTEGRATION: Compiler', () => {

--- a/waffle-compiler/test/compiler/e2e.ts
+++ b/waffle-compiler/test/compiler/e2e.ts
@@ -16,7 +16,9 @@ const configurations = [
   './test/projects/custom/config_promise.js',
   './test/projects/custom_solidity_4/config_solcjs.json',
   './test/projects/custom_solidity_4/config_docker.json',
-  './test/projects/custom/config_combined.js'
+  './test/projects/custom/config_combined.js',
+  './test/projects/solidity6/config_solcjs.json',
+  './test/projects/solidity6/config_docker.json',
 ];
 
 const artifacts = [

--- a/waffle-compiler/test/compiler/e2e.ts
+++ b/waffle-compiler/test/compiler/e2e.ts
@@ -18,7 +18,7 @@ const configurations = [
   './test/projects/custom_solidity_4/config_docker.json',
   './test/projects/custom/config_combined.js',
   './test/projects/solidity6/config_solcjs.json',
-  './test/projects/solidity6/config_docker.json',
+  './test/projects/solidity6/config_docker.json'
 ];
 
 const artifacts = [

--- a/waffle-compiler/test/projects/compilerOptions/config.json
+++ b/waffle-compiler/test/projects/compilerOptions/config.json
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
     "evmVersion": "constantinople"
-  }
+  },
+  "compilerVersion": "v0.5.9+commit.e560f70d"
 }

--- a/waffle-compiler/test/projects/custom/config.js
+++ b/waffle-compiler/test/projects/custom/config.js
@@ -2,5 +2,6 @@ module.exports = {
   name: "solidity 5, solcjs",
   sourceDirectory: "./test/projects/custom/custom_contracts",
   outputDirectory: "./test/projects/custom/custom_build",
-  nodeModulesDirectory: "./test/projects/custom/custom_node_modules"
+  nodeModulesDirectory: "./test/projects/custom/custom_node_modules",
+  compilerVersion: "v0.5.9+commit.e560f70d"
 };

--- a/waffle-compiler/test/projects/custom/config_combined.js
+++ b/waffle-compiler/test/projects/custom/config_combined.js
@@ -14,6 +14,7 @@ module.exports = {
         "": [ "ast" ]
       },
     }
-  }
+  },
+  compilerVersion: "v0.5.9+commit.e560f70d"
 };
 

--- a/waffle-compiler/test/projects/custom/config_promise.js
+++ b/waffle-compiler/test/projects/custom/config_promise.js
@@ -3,6 +3,7 @@ module.exports = new Promise((resolve, reject) => {
     name: "solidity 5, promise returned solcjs config",
     sourceDirectory: "./test/projects/custom/custom_contracts",
     outputDirectory: "./test/projects/custom/custom_build",
-    nodeModulesDirectory: "./test/projects/custom/custom_node_modules"
+    nodeModulesDirectory: "./test/projects/custom/custom_node_modules",
+    compilerVersion: "v0.5.9+commit.e560f70d"
   });
 });

--- a/waffle-compiler/test/projects/humanReadableAbi/config.json
+++ b/waffle-compiler/test/projects/humanReadableAbi/config.json
@@ -1,3 +1,4 @@
 {
-  "outputHumanReadableAbi": true
+  "outputHumanReadableAbi": true,
+  "compilerVersion": "v0.5.9+commit.e560f70d"
 }

--- a/waffle-compiler/test/projects/humanReadableAbi/config2.json
+++ b/waffle-compiler/test/projects/humanReadableAbi/config2.json
@@ -1,2 +1,3 @@
 {
+  "compilerVersion": "v0.5.9+commit.e560f70d"
 }

--- a/waffle-compiler/test/projects/invalidContracts/config_solcjs.json
+++ b/waffle-compiler/test/projects/invalidContracts/config_solcjs.json
@@ -2,5 +2,6 @@
   "name": "solidity 5, solcjs, invalid contract",
   "sourceDirectory": "./test/projects/invalidContracts",
   "outputDirectory": "./test/projects/invalidContracts/custom_build",
-  "nodeModulesDirectory": "./test/projects/invalidContracts/custom_node_modules"
+  "nodeModulesDirectory": "./test/projects/invalidContracts/custom_node_modules",
+  "compilerVersion": "v0.5.9+commit.e560f70d"
 }

--- a/waffle-compiler/test/projects/solidity6/config_docker.json
+++ b/waffle-compiler/test/projects/solidity6/config_docker.json
@@ -1,0 +1,8 @@
+{
+  "name": "solidity 6, docker",
+  "sourceDirectory": "./test/projects/solidity6/custom_contracts",
+  "outputDirectory": "./test/projects/solidity6/custom_build",
+  "nodeModulesDirectory": "./test/projects/solidity6/custom_node_modules",
+  "compilerType": "dockerized-solc",
+  "compilerVersion": "0.6.2"
+}

--- a/waffle-compiler/test/projects/solidity6/config_solcjs.json
+++ b/waffle-compiler/test/projects/solidity6/config_solcjs.json
@@ -1,0 +1,7 @@
+{
+  "name": "solidity 6, solcjs",
+  "sourceDirectory": "./test/projects/solidity6/custom_contracts",
+  "outputDirectory": "./test/projects/solidity6/custom_build",
+  "nodeModulesDirectory": "./test/projects/solidity6/custom_node_modules",
+  "compilerVersion": "v0.6.2+commit.bacdbe57"
+}

--- a/waffle-compiler/test/projects/solidity6/custom_contracts/Custom.sol
+++ b/waffle-compiler/test/projects/solidity6/custom_contracts/Custom.sol
@@ -1,0 +1,10 @@
+pragma solidity ^0.6.1;
+
+import "openzeppelin-solidity/CustomSafeMath.sol";
+
+
+contract Custom {
+  function dummy() public pure {
+
+  }
+}

--- a/waffle-compiler/test/projects/solidity6/custom_contracts/MyLibrary.sol
+++ b/waffle-compiler/test/projects/solidity6/custom_contracts/MyLibrary.sol
@@ -1,0 +1,8 @@
+pragma solidity ^0.6.1;
+
+
+library MyLibrary {
+  function sevenify(uint _number) public pure returns (uint) {
+    return _number + 7;
+  }
+}

--- a/waffle-compiler/test/projects/solidity6/custom_contracts/sub/One.sol
+++ b/waffle-compiler/test/projects/solidity6/custom_contracts/sub/One.sol
@@ -1,0 +1,17 @@
+pragma solidity ^0.6.1;
+
+import "openzeppelin-solidity/sub/ERC20.sol";
+
+
+contract One {
+  function dummy() public pure {
+
+  }
+}
+
+
+contract OneAndAHalf {
+  function someOtherDummy() public pure {
+
+  }
+}

--- a/waffle-compiler/test/projects/solidity6/custom_contracts/sub/Two.sol
+++ b/waffle-compiler/test/projects/solidity6/custom_contracts/sub/Two.sol
@@ -1,0 +1,17 @@
+pragma solidity ^0.6.1;
+
+import "./One.sol";
+import "../MyLibrary.sol";
+
+
+contract Two {
+  using MyLibrary for uint;
+
+  function dummy() public pure {
+
+  }
+
+  function useLibrary(uint _number) public pure returns (uint) {
+    return _number.sevenify();
+  }
+}

--- a/waffle-compiler/test/projects/solidity6/custom_node_modules/openzeppelin-solidity/CustomSafeMath.sol
+++ b/waffle-compiler/test/projects/solidity6/custom_node_modules/openzeppelin-solidity/CustomSafeMath.sol
@@ -1,0 +1,8 @@
+pragma solidity ^0.6.1;
+
+
+contract CustomSafeMath {
+  function dummy() public pure {
+
+  }
+}

--- a/waffle-compiler/test/projects/solidity6/custom_node_modules/openzeppelin-solidity/sub/ERC20.sol
+++ b/waffle-compiler/test/projects/solidity6/custom_node_modules/openzeppelin-solidity/sub/ERC20.sol
@@ -1,0 +1,8 @@
+pragma solidity ^0.6.1;
+
+
+contract ERC20 {
+  function dummy() public pure {
+
+  }
+}

--- a/waffle-provider/package.json
+++ b/waffle-provider/package.json
@@ -35,12 +35,11 @@
     "node": ">=10.0"
   },
   "dependencies": {
-    "@types/ganache-core": "^2.7.0",
-    "ethers": "^4.0.0",
-    "ganache-core": "2.9.0-istanbul.0"
+    "ethers": "^4.0.45",
+    "ganache-core": "^2.10.2"
   },
   "resolutions": {
-    "web3": "1.2.1"
+    "web3": "1.2.4"
   },
   "devDependencies": {
     "@types/chai": "^4.2.3",

